### ISSUE TITLE
Reduce the backoff cliff

### ISSUE THREAD

**Issue by thegcat** [matrix-org/synapse#14513]
### Description

Synapse currently has a pretty steep backoff cliff, meaning that if your homeserver has any issue longer than 1 day you have a small window to get back in federation after 6 days, and after that you're effectively defederated.

When synapse tries to contact another homeserver and encounters an error it will first retry the requests a few times and then go into an exponential backoff with a factor 5. There's a few more things involved, but for the sake of this discussion I'll start with a retry after 10 minutes and then backoffs that are 5 times longer each time. This is an oversimplification but is a close enough representation of our understanding of the phenomenon and illustration of the problem.

- 1st retry after 10 minutes
- 2nd retry after 50 minutes, total time since first error 60 minutes
- 3rd retry after 250 minutes, total time since first error 310 minutes (5 hours 10 minutes)
- 4th retry after 1250 minutes, total time since first error 1560 minutes (26 hours, roughly 1 day)
- 5th retry after 6250 minutes, total time since first error 7810 minutes (roughly 5,5 days)
- 6th retry after 31250 minutes, total time since first error 39060 minutes (a little over 27 days)

This effectively means that if your homeserver has an issue (in our case a [bad configuration](https://github.com/matrix-org/synapse/issues/14492#issuecomment-1321121551), but your hosting could be down, you're self-hosting at home and your internet goes down, whatever), and it takes more than 1 day to fix, you have another chance day 5 to get back on the federation but will still be defederated for a few days in-between, and after day 5 you could consider that you're effectively defederated, because a messenger with a lag of 2-3 weeks isn't worth much.

In our case we had the issue with just one of our homeservers and matrix.org, and were able to get the backoff reset before day 5, but this is sustainable neither for the admins of the "offending" homeserver that might need to contact each homeserver admin their homeserver federates with, nor for homeserver admins receiving requests to reset the backoff.

We also understand [removing the backoff cap was a deliberate decision](https://github.com/matrix-org/synapse/issues/5113), but would like to discuss reintroducing a "plateau" in the midst of the exponential backoff. Giving admins something like 2 weeks with frequent (once a day?) retries and 2 months with less frequent retries (every 3 days?) would add around 35 retries over 2 months before effectively defederating the offending homeserver.

### Steps to reproduce

- have an issue with your homeserver that looks like your homeserver doesn't federate anymore
- fix the issue after more than 1 day/more than 6 days
- be defederated 6 days/effectively forever

### Homeserver

matrix.fachschaften.org

### Synapse Version

1.71.0

### Installation Method

Other (please mention below)

### Platform

https://github.com/spantaleev/matrix-docker-ansible-deploy

### Relevant log output

```shell
None
```

**Comment by DMRobertson** [comment]
Thanks for your thorough write-up.

In https://github.com/matrix-org/synapse/issues/5113#issue-438402145, Rich writes:

> We would need to unblock such servers when we receive a valid request from them.

Does this logic/behaviour exist? E.g. if you send a message from your HS to a room that matrix.org is joined to, does matrix.org start sending new messages back to your HS?

**Comment by thegcat** [comment]
> In [#5113 (comment)](https://github.com/matrix-org/synapse/issues/5113#issue-438402145), Rich writes:
> 
> > We would need to unblock such servers when we receive a valid request from them.
> 
> Does this logic/behaviour exist? E.g. if you send a message from your HS to a room that matrix.org is joined to, does matrix.org start sending new messages back to your HS?

The observed behaviour at least doesn't suggest that this works. I can't say if it's implemented.

What we observed was that we had fixed the issue on our side on Saturday evening, but we started syncing up to matrix.org only on Monday evening after getting the backoff for our server reset on matrix.org through backchannels.

**Comment by DMRobertson** [comment]
> What we observed was that we had fixed the issue on our side on Saturday evening, but we started syncing up to matrix.org only on Monday evening after getting the backoff for our server reset on matrix.org through backchannels.

I'm not sure what those backchannels are, but Matrix.org had some issues with outbound federation to a limited number of homeservers on Monday. From Matrix.org logs, I can see that `fachschaften.org` was one of these destination HSes. This looks to be the same failure mode as https://github.com/matrix-org/synapse/issues/14492; closing this as a duplicate of that.

If you see this kind of failure again (no traffic from host, even after sending messages to it), feel free to reopen the issue and I'll take another look.
